vars/highLevelDataSetup.groovy

```diff
@@ -0,0 +1,23 @@
+#!groovy
+import uk.gov.hmcts.contino.Builder
+import uk.gov.hmcts.contino.PipelineCallbacksRunner
+import uk.gov.hmcts.contino.AppPipelineConfig
+import uk.gov.hmcts.contino.DockerImage
+
+def call(params) {
+  PipelineCallbacksRunner pcr = params.pipelineCallbacksRunner
+  AppPipelineConfig config = params.appPipelineConfig
+  Builder builder = params.builder
+  DockerImage.DeploymentStage deploymentStage = params.stage
+  def product = params.product
+
+  def dataSetupEnvironment = deploymentStage.label
+
+  if (config.highLevelDataSetup) {
```

Comment:
this check should be lifted higher to `withPipeline` that way we don't invoke this step at all if not required

Reply:
Here, I tried to handle it implicitly. So, when someone wants to use it, they don't need to handle if-check each time. I also tried it in my local Jenkins env, it is not invoked and not created a new stage in the pipeline at the moment.

Reply:
the step is still called though, the normal pattern that is follow is checks are done outside, if you create a new pipeline type, it would re-use this code and then you wouldn't want this feature flag to be internal to the method

Reply:
I though I can use the same flow that has been used in https://github.com/hmcts/cnp-jenkins-library/blob/dc214bce84171e1b983fb44069526bf4c8088b6d/vars/sectionDeployToAKS.groovy#L71. As I understand, sectionDeployToAKS has sub-stages and they are managed by flags implicitly at client project Jenkins file.

Reply:
That's not for flagging a whole step though, it's conditionally turning off part of the pipeline, see https://github.com/hmcts/cnp-jenkins-library/blob/master/vars/withPipeline.groovy#L83

Reply:
As discussed, we are leaving this as is